src/main/java/ch/njol/skript/expressions/ExprSpectatorTarget.java

```diff
@@ -24,60 +24,153 @@
+					"'spectator target of event-player'");
+			return false;
+		} else if (players == null && !getParser().isCurrentEvent(PlayerStartSpectatingEntityEvent.class, PlayerStopSpectatingEntityEvent.class)) {
+			Skript.error("The expression 'spectator target' may only be used in a start/stop/swap spectating target event");
+			return false;
+		}
+		return true;
 	}
 
 	@Override
-	public Class<? extends Entity> getReturnType() {
-		return Entity.class;
+	@Nullable
+	protected Entity[] get(Event event) {
+		if (EVENT_SUPPORT && players == null && !Delay.isDelayed(event)) {
+			if (event instanceof PlayerStartSpectatingEntityEvent) {
+				// Past state.
+				if (getTime() == -1)
+					return CollectionUtils.array(((PlayerStartSpectatingEntityEvent) event).getCurrentSpectatorTarget());
+				return CollectionUtils.array(((PlayerStartSpectatingEntityEvent) event).getNewSpectatorTarget());
+			} else if (event instanceof PlayerStopSpectatingEntityEvent) {
+				// There isn't going to be a future state in a stop spectating event.
+				if (getTime() == 1)
+					return new Entity[0];
+				return CollectionUtils.array(((PlayerStopSpectatingEntityEvent) event).getSpectatorTarget());
+			}
+		}
+		if (players == null)
+			return new Entity[0];
+		return players.stream(event).map(Player::getSpectatorTarget).toArray(Entity[]::new);
 	}
 
-	@Nullable
 	@Override
-	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
-		if (mode == Changer.ChangeMode.SET
-				|| mode == Changer.ChangeMode.RESET
-				|| mode == Changer.ChangeMode.DELETE) {
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		// Make 'spectator target' act as an entity changer. Will error in init for unsupported server platform.
+		if (players == null)
+			return DefaultChangers.entityChanger.acceptChange(mode);
+		if (mode == ChangeMode.SET || mode == ChangeMode.RESET || mode == ChangeMode.DELETE)
 			return CollectionUtils.array(Entity.class);
-		}
 		return null;
 	}
 
 	@Override
-	public void change(Event e, @Nullable Object[] delta, Changer.ChangeMode mode) {
-		for (Player player : getExpr().getArray(e)) {
-			if (player.getGameMode() == GameMode.SPECTATOR) {
-				switch (mode) {
-					case SET:
-						assert delta != null;
-						player.setSpectatorTarget((Entity) delta[0]);
-						break;
-					case RESET:
-					case DELETE:
-						player.setSpectatorTarget(null);
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		// Make 'spectator target' act as an entity changer. Will error in init for unsupported server platform.
+		if (players == null) {
+			Entity[] entities = get(event);
+			if (entities.length == 0)
+				return;
+			DefaultChangers.entityChanger.change(entities, delta, mode);
+			return;
+		}
+		switch (mode) {
+			case SET:
+				assert delta != null;
+				for (Player player : players.getArray(event)) {
+					if (player.getGameMode() != GameMode.SPECTATOR)
+						continue;
+					player.setSpectatorTarget((Entity) delta[0]);
```

Comment:
```suggestion
					if (player.getGameMode() == GameMode.SPECTATOR)
						player.setSpectatorTarget((Entity) delta[0]);
```
We can save a line :)